distributed/client.py

```diff
@@ -2959,8 +2984,21 @@ def expand_resources(resources):
                     out[tokey(kk)] = v
         return out
 
-    @staticmethod
-    def get_restrictions(collections, workers, allow_other_workers):
+    _expand_resources = _expand_per_key_specification
+
+    @classmethod
+    def _expand_retries(cls, retries, all_keys):
+        if retries and isinstance(retries, dict):
+            return cls._expand_per_key_specification(retries)
+        elif isinstance(retries, Integral):
+            # Each task unit may potentially fail, allow retrying all of them
+            return {name: retries for name in all_keys}
+        else:
+            raise TypeError("`retries` should be an integer or dict, got %r"
+                            % (type(retries,)))
```

Comment:
Would it be possible to have the exact same behavior for resources and retries?  I'm curious if there are reasons why the behaviors here should not be identical.

Reply:
Do you mean you don't want the `retries=3` shortcut in case you want the same value for all futures? Or something else?

Reply:
It seems like a very valuable shortcut to me.  I wonder if we might also have the same shortcut for resources as well like `resources={'GPU': 1}`.  I'm actually not sure what the current behavior for this is.

Reply:
So, to be clear, my objective here is consistent behavior across the different keywords more than an adherence to previous behaviors.

Reply:
Oh, I see. Do you want it as part of this PR?

Reply:
If you're up for it, yes.  We should probably try to maintain API consistency as part of the normal dev process.

Reply:
I'll do.

Reply:
Hmm, so there is semantic ambiguity here. If I pass `{'GPU': 1}` for several tasks at once, does each task need a GPU for itself, or do they only need one GPU for all of them? Probably the former, but just wanting to check.

Reply:
IMHO a logical semantic would be to have { ‘GPU’ : True } meaning that the task requires that there is a GPU, that could potentially be used by multiple tasks, whereas { ‘GPU’ : 1} would mean that the task required 1 GPU for its exclusive use. ( C.f { ‘DiskSpace’ : 1e9 } to mean a task needed 1GB of disk space for its exclusive use.)

Reply:
Presumably this same ambiguity exists for retries as well?

Reply:
@mrocklin I don't really see how.

Reply:
@rbubley `1 == True` means it may get confusing :-)

Reply:
@pitrou - pah! Don’t you want lots of isinstance(x, bool) to disambiguate?